pkg/tfbridge/info.go

```diff
@@ -529,10 +529,10 @@ func (m *MarshallableElem) Unmarshal() interface{} {
 		return nil
 	case m.Schema != nil:
 		return m.Schema.Unmarshal()
-	case m.Resource != nil:
-		return m.Resource.Unmarshal()
 	default:
-		return nil
+		// m.Resource might be nil in which case it was empty when marshalled. But Unmarshal can be called on
```

Comment:
Ugh what value were you getting here and in what context I'd love to scroll back a bit? This value must have:

```
m != nil
m.Schema == nil
m.Resource == nil
```

That seems really weird.

I have some case analysis here:

https://github.com/pulumi/pulumi-terraform-bridge/blob/228b97f2c51a58bf3077019d9c1c5e7690f66749/pkg/tfshim/shim.go#L79-#L79

You seem to be hitting something that's neither of the cases (in the MarshallableElem projection)

Reply:
Resource is just a map, and it's tagged 'omitempty' so a resource with no fields gets written out as `{}`, which then gets read back in as both fields being nil.

Reply:
There were some list elements in the Aws schema like that. On my phone so hard to go look them up, but I can check and reply tomorrow.

Reply:
```
                            "connector_profile_properties": {
                                "type": 5,
                                "required": true,
                                "element": {
                                    "resource": {
                                        "amplitude": {
                                            "type": 5,
                                            "optional": true,
                                            "element": {},
                                            "maxItems": 1
                                        },
```
That maps to an empty object in the codegen (see `ConnectorProfileConnectorProfileConfigConnectorProfilePropertiesAmplitude` type in the aws typescript sdk for example)